lib/syskit/network_generation/engine.rb

```diff
@@ -311,16 +311,6 @@ def find_selected_device_in_hierarchy(argument_name, leaf_task, requirements)
                 _, model, _ = leaf_task.requirements.resolved_dependency_injection.selection_for(nil, requirements)
                 if model && dev = model.arguments[argument_name]
                     return dev
-                else
-                    devices = Set.new
```

Comment:
I'm always for raising if there is an ambiguity, i.e. more or less what you propose:
 - if there are no choices, then don't do anything
 - if there is more than one choice, raise
 - if there is exactly one choice, assign